File: is52038b-labs/18/cpp/CompressedTrieQueryTest.cpp
#include <cppunit/extensions/HelperMacros.h>
#include <cppunit/CompilerOutputter.h>
#include <cppunit/extensions/TestFactoryRegistry.h>
#include <cppunit/ui/text/TestRunner.h>

#include "Trie.hpp"
#include "CompressedTrie.hpp"

class CompressedTrieQueryTest : public CppUnit::TestFixture {
  CPPUNIT_TEST_SUITE(CompressedTrieQueryTest);
  CPPUNIT_TEST(testCompressedQueryOne);
  CPPUNIT_TEST(testCompressedQueryTwo);
  CPPUNIT_TEST(testCompressedQueryCommonPrefix);
  CPPUNIT_TEST(testCompressedQueryStock);
  CPPUNIT_TEST_SUITE_END();
public:
  void setUp() { };
  void tearDown() { };

  void testCompressedQueryOne();
  void testCompressedQueryTwo();
  void testCompressedQueryCommonPrefix();
  void testCompressedQueryStock();
};

void CompressedTrieQueryTest::testCompressedQueryOne() {
  Trie *t = new Trie();
  t->insert("one");
  CompressedTrie *c = CompressedTrie::compressTrie(t);
  CPPUNIT_ASSERT(c->query("one"));
  CPPUNIT_ASSERT(!c->query("two"));
  CPPUNIT_ASSERT(!c->query("on"));
  CPPUNIT_ASSERT(!c->query("ones"));
}

void CompressedTrieQueryTest::testCompressedQueryTwo() {
  Trie *u = new Trie();
  Trie *t = u;
  t->insert("one");
  t->insert("two");
  CompressedTrie *c = CompressedTrie::compressTrie(t);
  CPPUNIT_ASSERT(c->query("one"));
  CPPUNIT_ASSERT(c->query("two"));
  CPPUNIT_ASSERT(!c->query("on"));
  CPPUNIT_ASSERT(!c->query("ones"));
}

void CompressedTrieQueryTest::testCompressedQueryCommonPrefix() {
  Trie *t = new Trie();
  t->insert("one");
  t->insert("only");
  CompressedTrie *c = CompressedTrie::compressTrie(t);
  CPPUNIT_ASSERT(c->query("one"));
  CPPUNIT_ASSERT(!c->query("two"));
  CPPUNIT_ASSERT(!c->query("on"));
  CPPUNIT_ASSERT(c->query("only"));
  CPPUNIT_ASSERT(!c->query("ones"));
}

void CompressedTrieQueryTest::testCompressedQueryStock() {
  Trie *t = new Trie();
  t->insert("bear");
  t->insert("bell");
  t->insert("bid");
  t->insert("bull");
  t->insert("buy");
  t->insert("sell");
  t->insert("stock");
  t->insert("stop");
  CompressedTrie *c = CompressedTrie::compressTrie(t);
  CPPUNIT_ASSERT(c->query("bear"));
  CPPUNIT_ASSERT(c->query("bell"));
  CPPUNIT_ASSERT(c->query("bid"));
  CPPUNIT_ASSERT(c->query("bull"));
  CPPUNIT_ASSERT(c->query("buy"));
  CPPUNIT_ASSERT(c->query("sell"));
  CPPUNIT_ASSERT(c->query("stock"));
  CPPUNIT_ASSERT(c->query("stop"));
  CPPUNIT_ASSERT(!c->query("be"));
  CPPUNIT_ASSERT(!c->query("bells"));
  CPPUNIT_ASSERT(!c->query("bide"));
  CPPUNIT_ASSERT(!c->query("hull"));
  CPPUNIT_ASSERT(!c->query("bay"));
  CPPUNIT_ASSERT(!c->query("sill"));
  CPPUNIT_ASSERT(!c->query("stack"));
  CPPUNIT_ASSERT(!c->query("stops"));
}

CPPUNIT_TEST_SUITE_NAMED_REGISTRATION(CompressedTrieQueryTest, "CompressedQuery");
CPPUNIT_REGISTRY_ADD_TO_DEFAULT("CompressedQuery");

File: completed-labs/18/java/CompressedTrieQueryTest.java
import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNull;
import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertTrue;
import java.util.Random;
import org.junit.Test;

public class CompressedTrieQueryTest {
    @Test
    public void testCompressedQueryOne() {
	Trie t = new Trie();
	t.insert("one");
	CompressedTrie c = CompressedTrie.compressTrie(t);
	assertTrue(c.query("one"));
	assertTrue(!c.query("two"));
	assertTrue(!c.query("on"));
	assertTrue(!c.query("ones"));
    }
    @Test
    public void testCompressedQueryTwo() {
	Trie t = new Trie();
	t.insert("one");
	t.insert("two");
	CompressedTrie c = CompressedTrie.compressTrie(t);
	assertTrue(c.query("one"));
	assertTrue(c.query("two"));
	assertTrue(!c.query("on"));
	assertTrue(!c.query("ones"));
    }
    @Test
    public void testCompressedQueryCommonPrefix() {
	Trie t = new Trie();
	t.insert("one");
	t.insert("only");
	CompressedTrie c = CompressedTrie.compressTrie(t);
	assertTrue(c.query("one"));
	assertTrue(!c.query("two"));
	assertTrue(!c.query("on"));
	assertTrue(c.query("only"));
	assertTrue(!c.query("ones"));
    }
    @Test
    public void testCompressedQueryStock() {
	Trie t = new Trie();
	t.insert("bear");
	t.insert("bell");
	t.insert("bid");
	t.insert("bull");
	t.insert("buy");
	t.insert("sell");
	t.insert("stock");
	t.insert("stop");
	CompressedTrie c = CompressedTrie.compressTrie(t);
	assertTrue(c.query("bear"));
	assertTrue(c.query("bell"));
	assertTrue(c.query("bid"));
	assertTrue(c.query("bull"));
	assertTrue(c.query("buy"));
	assertTrue(c.query("sell"));
	assertTrue(c.query("stock"));
	assertTrue(c.query("stop"));
	assertTrue(!c.query("be"));
	assertTrue(!c.query("bells"));
	assertTrue(!c.query("bide"));
	assertTrue(!c.query("hull"));
	assertTrue(!c.query("bay"));
	assertTrue(!c.query("sill"));
	assertTrue(!c.query("stack"));
	assertTrue(!c.query("stops"));
    }
}
